tasks.c

```diff
@@ -855,7 +856,7 @@ static void prvAddNewTaskToReadyList( TCB_t * pxNewTCB ) PRIVILEGED_FUNCTION;
             portGET_TASK_LOCK();
             portGET_ISR_LOCK();
 
-            portSET_CRITICAL_NESTING_COUNT( uxPrevCriticalNesting );
+            portSET_CRITICAL_NESTING_COUNT( xCoreID, uxPrevCriticalNesting );
```

Comment:
The running state of the task may be altered after enabling or disabling interrupts. We need to obtain the core ID again to determine which core the task is currently running on.

Reply:
Done!